api/tests/unit/config/fixtures/data_fixtures/AppFixturesTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests\unit\config\fixtures\data_fixtures;
+
+use App\DataFixtures\AppFixtures;
+use App\Entity\Question;
+use App\Entity\Quiz;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+final class AppFixturesTest extends KernelTestCase
+{
+
+    /** @var \Doctrine\ORM\EntityManager */
+    private $entityManager;
+
+    protected function setUp(): void
+    {
+        $kernel = self::bootKernel();
+
+        $this->entityManager = $kernel->getContainer()
+            ->get('doctrine')
+            ->getManager();
+    }
+
+    public function invokeMethod(object &$object, string $methodName, array $parameters = array())
+    {
+        $reflection = new \ReflectionClass(get_class($object));
+        $method = $reflection->getMethod($methodName);
+        $method->setAccessible(true);
+
+        return $method->invokeArgs($object, $parameters);
+    }
+
+    public function invokeProperty(object &$object, string $propertyName, mixed $parameter)
+    {
+        $reflection = new \ReflectionClass(get_class($object));
+        $property = $reflection->getProperty($propertyName);
+        $property->setAccessible(true);
+        $property->setValue($object, $parameter);
+
+        return $object;
+    }
+
+    // TODO : ask Peter : docker-compose exec api bin/console --env=test doctrine:database:create
+    public function testLoad() {
```

Comment:
@pfwd here at this point need help 🆘

Reply:
I've poorly written this data fixture as its not easily unit testable.  After reviewing the code I think it needs some refactoring. I would prefer to split unit tests and integration tests. I don't think we should be coupling the database calls to this test.

I suggest for the unit test:
- Set `ObjectManager` as a class property during load. Unfortunately Doctrines `FixtureInterface` requires it a property of `load`. So perhaps set it there. We can then mock the `ObjectManager` in the tests via a setup method.
-  Change the method visibility scopes from `protected` to `public` so they can be called via test methods independently from the `load` method.
- Mock the `ObjectManager` methods `persist` and `flush`.  We shouldn't be testing those
- Restructure the `load` method into smaller methods that only have one loop each giving them a single responsibility. 

I suggest for the integration test (Another ticket):
- Then create a command test for the full `bin/console --env=test doctrine:database:create` command. 
- This should run that command and then check the database for saved records.

I'm open to further suggestions on how to make this more testable.

Reply:
@pfwd I tried the cover your suggestions

- Please share with me if we should add/update anything else, nothing else comes to my mind currently :)